gmso/external/convert_parmed.py

```diff
@@ -426,6 +462,65 @@ def _dihedral_types_from_pmd(structure, dihedral_types_member_map=None):
     return pmd_top_dihedraltypes
 
 
+def _improper_types_from_pmd(structure, improper_types_member_map=None):
+    """Convert ParmEd improper types to GMSO ImproperType.
+
+    This function take in a Parmed Structure, iterate through its
+    improper_types and dihedral_types with the `improper=True` flag,
+    create a corresponding GMSO.ImproperType, and finally return
+    a dictionary containing all pairs of pmd.ImproperType
+    (or pmd.DihedralType) and GMSO.ImproperType
+
+    Parameters
+    ----------
+    structure: pmd.Structure
+        Parmed Structure that needed to be converted.
+    improper_types_member_map: optional, dict, default=None
+        The member types (atomtype string) for each atom associated with the improper_types the structure
+
+    Returns
+    -------
+    pmd_top_impropertypes : dict
+        A dictionary linking a pmd.ImproperType or pmd.DihedralType
+        object to its corresponding GMSO.ImproperType object.
+    """
+    pmd_top_impropertypes = dict()
+    improper_types_member_map = _assert_dict(
+        improper_types_member_map, "improper_types_member_map"
+    )
+
+    for dihedraltype in structure.dihedral_types:
+        improper_params = {
+            "k": (dihedraltype.phi_k * u.Unit("kcal / mol")),
+            "phi_eq": (dihedraltype.phase * u.degree),
+            "n": dihedraltype.per * u.dimensionless,
+        }
+        expr = gmso.ImproperType._default_potential_expr()
+        expr.parameters = improper_params
+        member_types = improper_types_member_map.get(id(dihedraltype))
+        top_impropertype = gmso.ImproperType(
+            potential_expression=expr, member_types=member_types
+        )
+        pmd_top_impropertypes[dihedraltype] = top_impropertype
+
+    for impropertype in structure.impropers:
+        improper_params = {
+            "xi_k": (impropertype.psi_k * u.Unit("kcal/mol")),
+            "xi_eq": (impropertype.psi_eq * u.Unit("kcal/mol")),
+        }
+
+        member_types = improper_types_member_map.get(id(impropertype))
+
+        top_impropertype = gmso.ImproperType(
```

Comment:
One quick thought or question/thought here. Can we use the same/uniform expression throughout the conversion?

Here's what we have defined for a standard HarmonicImproperPotential to look like.
https://github.com/mosdef-hub/gmso/blob/3a387c707109e64b5f178591249db94a134d2ac4/gmso/lib/jsons/HarmonicImproperPotential.json#L1-L5

Reply:
After chatting with @bc118, I think this makes sense instead of creating a new custom potential.